test/integration/tags/inline_comment_test.rb

```diff
@@ -5,8 +5,35 @@
 class InlineCommentTest < Minitest::Test
   include Liquid
 
-  def test_tag
+  def test_tag_in_different_styles
     assert_template_result('', '{% # This text gets ignored %}')
+    assert_template_result('', '{%# This text gets ignored #%}')
+    assert_template_result('', '{%# This text gets ignored %}')
+    assert_template_result('', '{%#- This text gets ignored -#%}')
+  end
+
+  def test_test_syntax_error
+    assert_template_result('fail', '{% #This doesnt work %}')
+
+    assert false 
+  rescue 
+    # ok good
+  end
+
+  def test_tag_ws_stripping
+    assert_template_result('', '   {%#- This text gets ignored -#%}    ')
+  end
+
+  def test_comment_inline_tag
+    assert_template_result('ok', '{% echo "ok" # output something from a tag %}')
```

Comment:
Worth noting that this removes a lot of the benefit of treating `#` as its own tag, because it means the concern leaks into the syntax of other tags too.

This also has implications for the API, since we'd have to either
1. Do pre-processing on all tag input before passing it, which is a breaking change in a strict sense
2. Only support it for our own tags, which could lead to inconsistencies for other clients.

The second of these may still be the right call, but special-casing those tags could be more complex in Liquid-C.

Reply:
What exactly is the benefit of treating it as a tag? My (limited!) experience with parser creation suggests that comments are usually dropped very early in the process of parsing/lexing and this feels like the right approach for liquid as well.